### ISSUE TITLE
Mini Cart: Add contrast checking for Price, Icon and Product Count color settings

### ISSUE THREAD

**Issue by danieldudzic** [woothemes/woocommerce#42350]
woocommerce/woocommerce-blocks#9647 adds Price, Icon and Product Count color settings.

As @tjcafferkey pointed in https://github.com/woocommerce/woocommerce-blocks/pull/9647#issuecomment-1572239688 it would be wise to add contrast checking.

**Comment by amesplant** [comment]
I am looking into setting up a contrast checker when setting colors

**Comment by Manussakis** [comment]
Hey @tjcafferkey 

I am looking [at your comment](https://github.com/woocommerce/woocommerce-blocks/pull/9647#issuecomment-1572239688) that led to this issue and it seems to me we need design inputs before starting the development.

Some questions for designers are:
- Where do we want to put the messages?
- Should success messages be displayed or only the failed ones?
- What colors and icons should we use for success and fail messages?
- What the message texts should read?

As for development, the steps I am considering are:
- Once a color is selected, get the first mini-cart's parent element with a background color set.
- Check if the contrast ratio is of at least 4.5:1 (https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html)
- Display the contrast check message.
- If the colors are reset, hide the message.

**Comment by vikrampm1** [comment]
Moving this to our backlog until the designs are ready.

**Comment by Manussakis** [comment]
@elizaan36 Could you please review this issue and share your design inputs?
Thanks!

cc: @vikrampm1

**Comment by pierorocca** [comment]
This should be a universal feature on any block that allows setting of text and background colors? i.e. why is this only a mini-cart issue? @ralucaStan happy to start in the mini-cart if we have a design that will be used and implemented identically in every applicable cart. Otherwise this is probably better served with the Storefront team to set the standard since they cover more surface area related to style settings.

**Comment by elizaan36** [comment]
This would be a huge win as a universal feature for sure, not sure the process to do that but I agree we should!

Here's one possibility - 

Present an error upon detection of the color contrast issue.
<img width="804" alt="Image" src="https://github.com/user-attachments/assets/5e65ab2c-3acb-4376-b704-e00f154a35c3" />

Tooltip to inform the user and from there they can make any desired adjustments.
<img width="804" alt="Image" src="https://github.com/user-attachments/assets/7a917464-5082-46f3-ab61-7685f1f69aa2" />

**Comment by ralucaStan** [comment]
There is a Gutenberg component that can be used for this https://wordpress.github.io/gutenberg/?path=/docs/blockeditor-contrastchecker--docs

**Comment by pierorocca** [comment]
Thanks @ralucaStan! @elizaan36 mind checking that out and commenting if that's the appropriate component to use? Obviously we'd like to standardize if it makes sense to do so.

**Comment by Manussakis** [comment]
Thanks, @elizaan36 for the design inputs!
Thanks, @ralucaStan for sharing the `<ContrastChecker />` component!

It would be great if we could leverage the `<ContrastChecker />` component and standardize this kind of warning regardless of whether editors edit default WP or WooCommerce blocks.

I noticed that there are three different contrast color messages, depending on the foreground/background color combination, which means each message needs to be related to its input, i.e., Price, Icon, and Product Count.

I played a bit with the code and wanted to share with you what I came up with:

https://github.com/user-attachments/assets/fa309c26-38ac-4f33-b96f-e8abf1d9cbda

@elizaan36 You will notice that I didn't add the warning icon, that's because, based on my investigation, this color picker component is an [experimental one](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/editor-components/color-panel/index.tsx#L13), that only accepts text as label. I can look into it even further if the icon is important, but we might not want it to appear if we want to align with how WP displays the contrast color message by default (without the icon).

Please watch the attached video and let me know your thoughts.

Thanks!

**Comment by pierorocca** [comment]
Thanks @Manussakis for the visuals. That's very much appreciated. Looks really good to me and the instant, visible notification does a great job at grabbing attention.

**Comment by ralucaStan** [comment]
@elizaan36 the same component is used in Product filters block, for the price filter

**Comment by elizaan36** [comment]
Wonderful that exists already. Yes, the notice works as-is. It could appear more in context of the actual color picker rather than below but I think it does accomplish the objective of notifying the user on what the problem is and what to do about it.

**Comment by Manussakis** [comment]
Thanks @elizaan36 !
I will go ahead and polish the implementation code to send it to code review.
